Sources/protoc-gen-grpc-swift/Docs.docc/spm-plugin.md

```diff
@@ -0,0 +1,113 @@
+
+1. If you are on MacOS, installing it via `brew install protoc`
+2. Download the binary from [Google's github repository](https://github.com/protocolbuffers/protobuf).
+
+### Adding the proto files to your target
+
+Next, you need to add the `.proto` files for which you want to generate your Swift types to your target's
+source directory. You should also commit these files to your git repository since the generated types
+are now generated on demand.
+
+> Note: imports on your `.proto` files will have to include the full path from the target source to the `.proto` file you wish to import.
+
+### Adding the plugin to your manifest
+
+After adding the `.proto` files you can now add the plugin to the target inside your `Package.swift` manifest.
+First, you need to add a dependency on `grpc-swift`. Afterwards, you can declare the usage of the plugin
+for your target. Here is an example snippet of a `Package.swift` manifest:
+
+```swift
+let package = Package(
+  name: "YourPackage",
+  products: [...],
+  dependencies: [
+    ...
+    .package(url: "https://github.com/grpc/grpc-swift", from: "1.9.1"),
+    ...
+  ],
+  targets: [
+    ...
+    .executableTarget(
+        name: "YourTarget",
+        plugins: [
+            .plugin(name: "SwiftGRPCPlugin", package: "grpc-swift")
+        ]
+    ),
+    ...
+)
+
+```
+
+### Configuring the plugin
+
+Lastly, after you have added the `.proto` files and modified your `Package.swift` manifest, you can now
+configure the plugin to invoke the `protoc` compiler. This is done by adding a `swift-grpc-config.json`
+to the root of your target's source folder. An example configuration file looks like this:
+
+```json
+{
+    "invocations": [
+        {
+            "protoFiles": [
+                "Path/To/Foo.proto",
+            ],
+            "visibility": "internal"
+        },
+        {
+            "protoFiles": [
+                "Bar.proto"
+            ],
+            "visibility": "public"
+        }
+    ]
+}
+
+```
+
+In the above configuration, you declared two invocations to the `protoc` compiler. The first invocation
+is generating Swift types for the `Foo.proto` file with `internal` visibility. The second invocation
+is generating Swift types for the `Bar.proto` file with the `public` visibility.
+
+> Note: paths to your `.proto` files will have to include the full path from the target source to the `.proto` file location. 
+
+### Defining the path to the protoc binary
+
+The plugin needs to be able to invoke the `protoc` binary to generate the Swift types. 
+There are two ways how this can be achieved. First, by default, the package manager looks into
+the `$PATH` to find binaries named `protoc`. This works immediately if you use `swift build` to build
+your package and `protoc` is installed in the `$PATH` (`brew` is adding it to your `$PATH` automatically).
+However, this doesn't work if you want to compile from Xcode since Xcode is not passed the `$PATH`.
```

Comment:
This should also work (not tested): alternatively you can start Xcode from a terminal, which will make Xcode inherit the environment from the shell.

```zsh
$ xed Package.swift
```

Reply:
Oh you are right. This is great that means besides adding the `PROTOC_PATH` env variable we can also use the `PATH`. This is something we should document in both gRPC-swift and swift-protobuf. For the people that have protoc already in their `$PATH` it is way easier.

Reply:
Thanks Tobias, I'll modify the docs slightly to include this option. We should also modify the Protobuf plugin docs.